main/lsp/ErrorReporter.h

```diff
@@ -0,0 +1,37 @@
+#ifndef RUBY_TYPER_LSP_ERRORREPORTER_H
+#define RUBY_TYPER_LSP_ERRORREPORTER_H
+
+#include "ast/ast.h"
+#include "core/core.h"
+#include "main/lsp/LSPConfiguration.h"
+
+namespace sorbet::realmain::lsp {
+
+struct ErrorStatus {
+    // The epoch at which we last sent diagnostics for this file.
+    u4 sentEpoch;
```

Comment:
Here's a weird C++-ism... for primitive types (e.g. numbers and booleans), you should always provide a default initialization value (either in a constructor or in the definition). Otherwise, they will be initialized to random garbage.

```suggestion
    u4 sentEpoch = 0;
```

The random garbage comes into play if you ever construct one of these structs without specifying field values, e.g.:

```c++
ErrorStatus a;
// a.sentEpoch might be anything!
```

Reply:
"Why are they initialized to random garbage?" you might ask?

Well, at a low level, when you ask the OS for n bytes of memory so that you can create an object, it'll mark n bytes as yours and hand them to you. But they will contain the last thing written to them by whatever other program happened to be using them last.

The sane thing to do is to zero initialize that memory (which would make `sentEpoch` `0` and `hasErrors` `false`), but C++ doesn't do this since it adds overhead. 🤷 

Now, if this field were a class type, C++ would initialize it by calling the class's "default constructor" if it has one (which is the class's constructor with no arguments). If it doesn't have one, the compiler will complain.